okio-files/src/jsMain/kotlin/okio/NodeJsSystemFilesystem.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import fs.Dirent
+import fs.MakeDirectoryOptions
+import fs.mkdirSync
+import fs.openSync
+import fs.opendirSync
+import fs.statSync
+import okio.Path.Companion.toPath
+
+/**
+ * Use [Node.js APIs][node_fs] to implement the Okio filesystem interface.
+ *
+ * This class needs to make calls to some fs APIs that have multiple competing overloads. To
+ * unambiguously select an overload this passes `undefined` as the target type to some functions.
+ *
+ * [node_fs]: https://nodejs.org/dist/latest-v14.x/docs/api/fs.html
+ */
+object NodeJsSystemFilesystem : Filesystem() {
```

Comment:
This should probably have the word "Sync" in it so callers are aware that it does not use non-blocking APIs.

Reply:
Oh, I should have made this `internal`, users won’t ever interact directly with this. But we should document `Filesystem` as synchronous / blocking.